job-runner/chain/chain.go

```diff
@@ -92,10 +65,8 @@ func (c *Chain) NextJobs(jobId string) proto.Jobs {
 	return nextJobs
 }
 
-// IsRunnable returns whether or not a job is runnable. A job is considered
-// runnable if it is Pending or Stopped with some retry attempts remaining,
-// and all of its previous jobs are complete. If any previous jobs are not
-// complete, the job is not runnable.
+// IsRunnable returns true if the job is runnable. A job is runnable if its
+// state is PENDING or STOPPED and all immediately previous jobs are state COMPLETE.
 func (c *Chain) IsRunnable(jobId string) bool {
```

Comment:
Not directly related to this PR, but why do we even have the private isRunnable when all the public IsRunnable does it call isRunnable?

Reply:
`Chain.IsDoneRunning` calls `Chain.isRunnable`.  The caller locks whatever, and `isRunnable` does not lock anything.